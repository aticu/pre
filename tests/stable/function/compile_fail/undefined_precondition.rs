use pre::pre;

fn foo() {}

#[pre]
fn main() {
    #[assure("is bar", reason = "foo is bar")]
    foo()
}
