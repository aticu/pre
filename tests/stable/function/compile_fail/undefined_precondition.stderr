error[E0574]: expected struct, variant or union type, found function `foo`
 --> $DIR/undefined_precondition.rs:8:5
  |
8 |     foo()
  |     ^^^ not a struct, variant or union type

error[E0061]: this function takes 0 arguments but 1 argument was supplied
 --> $DIR/undefined_precondition.rs:8:5
  |
3 |   fn foo() {}
  |   -------- defined here
...
7 | /     #[assert_pre(condition("must be bar", reason = "is bar"))]
8 | |     foo()
  | |     ^^-
  | |_____|_|
  |       | supplied 1 argument
  |       expected 0 arguments
